lib/posix-process/events.c

```diff
@@ -0,0 +1,161 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2024, Unikraft GmbH and The Unikraft Authors.
+ * Copyright (c) 2022, NEC Laboratories Europe GmbH, NEC Corporation.
+ *                     All rights reserved.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <uk/init.h>
+#include <uk/process.h>
+
+UK_EVENT(POSIX_PROCESS_CLONE_EVENT);
+UK_EVENT(POSIX_PROCESS_EXECVE_EVENT);
+UK_EVENT(POSIX_PROCESS_PPEXIT_EVENT); /* pprocess */
+UK_EVENT(POSIX_PROCESS_PTEXIT_EVENT); /* pthread */
+UK_EVENT(POSIX_PROCESS_RELEASE_EVENT);
+
+extern const struct posix_process_clonetab_entry _posix_process_clonetab_start[];
+extern const struct posix_process_clonetab_entry _posix_process_clonetab_end;
+
+#define posix_process_clonetab_foreach(itr)				\
+	for ((itr) = DECONST(struct posix_process_clonetab_entry*,	\
+			     _posix_process_clonetab_start);		\
+	     (itr) < &(_posix_process_clonetab_end);			\
+	     (itr)++)
+
+static __uk_tls struct {
+	bool is_cloned;
+	__u64 cl_flags;
+} cl_status = { false, 0x0 };
+
+/* clone() flags we can handle. We initialize these to the flags
+ * we manage internally in libposix-process and during init append
+ * the flags handlers have registered for.
+ */
+__u64 cl_flags_handled = CLONE_SETTLS | CLONE_VFORK | CLONE_DETACHED | CLONE_VM |
```

Comment:
I think `cl_flags_handled` and `cl_flags_handled_set` could probably also be static.Unless of course you did intend for it to be callable within the lib (since you did not add it in exportsysms.uk), though I highly doubt it for the latter which is in inittab.